### PR TITLE
Add new resource: `aws_vpc_endpoint_policy`

### PR DESCRIPTION
**Pull request by julienduchesne**
Signed-off-by: Julien Duchesne <julien.duchesne@grafana.com>

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by sergenyalcin** [comment]
@julienduchesne Could you please copy the generated examples to examples folder:

`examples-generated/ec2/vpcendpointpolicy.yaml` -(copy)> `examples/ec2/vpcendpointpolicy.yaml`

Then we'll run the automated tests by using the example

**Comment by julienduchesne** [comment]
> @julienduchesne Could you please copy the generated examples to examples folder:
> 
> `examples-generated/ec2/vpcendpointpolicy.yaml` -(copy)> `examples/ec2/vpcendpointpolicy.yaml`
> 
> Then we'll run the automated tests by using the example

Done!

**Comment by sergenyalcin** [comment]
/test-examples="examples/ec2/vpcendpointpolicy.yaml"

File: examples/ec2/vpcendpointpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  forProvider:
+    region: us-west-1
+    serviceNameSelector:
+      matchLabels:
+        testing.upbound.io/example-name: aws_vpc_endpoint_service
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: example
+
+---
```
Comment (sergenyalcin) [inline examples/ec2/vpcendpointpolicy.yaml:68]: Could you please remove this `---`? This statement at the end of the file causes the end-to-end tests to fail.

**Comment by ulucinar** [comment]
Hi @julienduchesne,
Looks like we are missing a dependency in the example manifests in `examples/ec2/vpcendpointpolicy.yaml` as the logs from the uptest's [cluster-dump](https://github.com/upbound/provider-aws/suites/10283735528/artifacts/505190899) reveals:
```shell
...
1.6733555292646885e+09	DEBUG	provider-aws	Cannot resolve managed resource references	{"controller": "managed/ec2.aws.upbound.io/v1beta1, kind=vpcendpoint", "request": "/example", "uid": "3be30ebf-8757-442b-b222-e874e46c4da3", "version": "3697", "external-name": "", "error": "cannot resolve references: mg.Spec.ForProvider.ServiceName: no resources matched selector"}
...
```

Could you please add the necessary dependencies to get the the example manifest working?

**Pull request closed without merging**